main.c

```diff
@@ -125,6 +124,44 @@ void updateDisplay(void const *arg) {
 
 void gameLogic(void const *arg) {
 		blinkLEDs = false;
+	// Array of enemies, each with an x and y coordinates, when an enemy is hit it changes the y coordinate to above the screen 
+	// so it can fall again.
+	// number of enemies will always stay the same but their speed of descent will increase the longer the player is alive.
+	// for loop iterates through all the enemies, checks if they are below the bottom (player dies) checks for collision with the bullets
+	// there will only be one to three bullet object. The fire rate will be such that only 1-3 bullets will be on the screen at a time. 
+	// a semaphore can be used to make sure that display is only done after all collision detection is complete. This might not be neccesary.
+	// score will go up with time and with enemies killed
+	// bullet struct has position var and fired var (is not drawn if not fired)
+	// enemy struct has position var and death timer var (controls the amount of time between death and respawn) 
+	
+	// note: checking each enemy for bullet collision is inefficient but doesnt matter so whatever
+	// sudo code: 
+	// player.x  += player_direction* playerSpeed
+	//
+	// // shooting
+	// if buttonPressed && fireRateTimer < 1
+	//	fireRateTimer = fire Rate
+	//	bullet position = player position
+	//	bullet.fired = true
+	//
+	// if fireRateTimer > 0
+	//	fireRateTimer --
+	//
+	// // enemies and collision detection
+	// for all enemies 
+	//	if enemy.deathTimer > 0
+	//		enemy.deathTimer --
+	//		continue
+	//	enemy.ypos -= falling speed
```

Comment:
Hey Kenta, I think you should add something for when the enemy reaches the bottom of the screen